design-proposals/eim-platform-bundle-improvements.md

```diff
@@ -0,0 +1,133 @@
+
+**EIM:**
+
+- Sets the sole ownership of EIM cloud init template to EIM codebase.
+  It is NOT possible to overwrite EIM cloud-init template via the Platform Bundle,
+  because EIM cloud-init is foundational piece that, if mis-configured, may make the entire EMF unfunctional.
+- It is still possible to overwrite EIM Installer script for Ubuntu via the Platform Bundle.
+  This may be needed if we want support other OS distros in the future.
+- Extracts EIM cloud-init template to a separate sub-module of infra-onboarding.
+  The new sub-module is versioned separately following SemVer.
+  EIM cloud-init template is published to Release Service as version OCI artifact.
+- OM imports EIM cloud-init template as versioned Go module.
+- EIM extends the implementation of Platform Bundle to supply provisioned OS
+  with arbitrary files defined as part of the Platform Bundle.
+  It implies change of the platform bundle YAML format.
+- `platformBundle` field of the OS resource should become R/W field from northbound API,
+  so that external users can create their own custom OS profiles with arbitrary files packages.
+
+**EEF:**
+
+- EEF downloads EIM cloud-init template from Release Service for any EEF certification process.
+  EIM cloud-init template is not curated by EEF.
+- EEF should implement CI/CD to track any changes to Installer script and
+  generate a new version of the curated Installer script if any change introduced.
+
+### OS-level files as part of Platform Bundle
+
+The proposal extends the definition of Platform Bundle to provide additional OS-level files as follows:
+
+```yaml
+platformBundle:
+  files:
+    - type: TARBALL
+      artifact_type: OCI
+      location: registry-rs.edgeorchestration.intel.com/edge-orch/en/files/platformbundle/standalone-edge-node:0.1.1
+      target_path: /
+    - type: RAW
+      artifact_type: NON_OCI
+      location: https://raw.githubusercontent.com/open-edge-platform/infra-onboarding/refs/heads/main/onboarding-manager/pkg/platformbundle/ubuntu-22.04/Installer
+      target_path: /opt/intel_edge_node/Installer
+```
+
+The entire `platformBundle` field is converted to JSON as stored as `platform_bundle` field of OS resource.
+Onboarding Manager is responsible for reading internal structure and generating Tinkerbell workflow accordingly.
+There should be a new Tinker action that downloads files from RS and writes them to target OS location.
+
+## Rationale
+
+N/A
+
+## Affected components and Teams
+
+EIM - OS profiles, Onboarding Manager, Tinker actions
+
+- Onboarding Manager:
+  - Extract cloud-init template to a separate Go sub-module in infra-onboarding with CI/CD and SemVer versioning
+  - Extend Platform Bundle library to handle new YAML format
+  - Extend Tinkerbell workflows with action to download Platform Bundle files and write them to target OS file system.
+- Tinker actions:
+  - Implement new Tinker action that downloads Release Service artifacts (can be non-OCI to avoid implementing `oras`)
```

Comment:
Since we will now be supporting non-OCI artifacts in the release service as part of this, the Tinker action needs to detect if the artifact being requested is non-OCI or not before attempting to pull. Is there already an implementation for this or will it have to be added as part of the new Tinker action?

Reply:
We should pass information, along with the URL, on whether a file is OCI or non-OCI.

Reply:
Tinker actions would need to have logic for both. There is already implementation supporting oras and non-OCI downloads but it's distributed over different tinker actions so some code refactoring would be required.